nav2_behavior_tree/plugins/action/remove_passed_goals_action.cpp

```diff
@@ -31,12 +31,44 @@ RemovePassedGoals::RemovePassedGoals(
   viapoint_achieved_radius_(0.5)
 {
   getInput("radius", viapoint_achieved_radius_);
-
-  getInput("global_frame", global_frame_);
-  getInput("robot_base_frame", robot_base_frame_);
+  
   tf_ = config().blackboard->get<std::shared_ptr<tf2_ros::Buffer>>("tf_buffer");
   auto node = config().blackboard->get<rclcpp::Node::SharedPtr>("node");
   node->get_parameter("transform_tolerance", transform_tolerance_);
+
+  if(!getInput("global_frame", global_frame_)) {
+    RCLCPP_INFO(node->get_logger(), "Parameter 'global_frame' not provided by behavior tree xml file, trying to get it from ros2 parameter file");
+    if(!node->get_parameter("global_frame", global_frame_)){
+      global_frame_ = "map";
+      RCLCPP_INFO(node->get_logger(), "Parameter 'global_frame' not provided by ros2 parameter file, using default value '%s'",
+        global_frame_.c_str());
+    }
+    else{
+      RCLCPP_INFO(node->get_logger(), "Parameter 'global_frame' provided by ros2 parameter file, using value '%s'",
+        global_frame_.c_str());
+    }
+  }
+  else{
+    RCLCPP_INFO(node->get_logger(), "Parameter 'global_frame' provided by behavior tree xml file, using value '%s'",
+      global_frame_.c_str());
+  }
+
+  if(!getInput("robot_base_frame", robot_base_frame_)) {
```

Comment:
This logic for `robot_base_frame` and `global_frame` is the same. So if you make the function, you can also make 1 function that both of these checks share if you make the actual parameter to check on a parameter of the method. E.g. 

```
param_value  deconflictPortAndParamFrame(node, param_name, this)
```

`this` is so that you can call `getInput`

Reply:
Added this function to bt_utils.hpp